src/Algebra/Morphism/TwoMagmas.agda

```diff
@@ -0,0 +1,52 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some properties of Magma homomorphisms
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+
+open import Algebra using (Magma)
+import Algebra.Morphism.Definitions as Definitions
+open import Data.Product using (_,_)
+open import Function using (id; _∘_)
+open import Function.Definitions using (Inverseᵇ)
+open import Relation.Binary using (_Preserves_⟶_)
+import Relation.Binary.Reasoning.Setoid as EqR
+open import Relation.Unary using (Pred)
```

Comment:
All non-necessary imports should go inside the module.

Reply:
I change this head to                                                                       
```                                                                                         
open import Algebra using (Magma)                                                           
                                                                                            
module Algebra.Morphism.Consequences {α α= β β=} (M : Magma α α=) (M' : Magma β β=)         
  where                                                                                     
import Algebra.Morphism.Definitions as Definitions                                          
open import Data.Product using (_,_)                                                        
...                                                                                                                                   
import Relation.Binary.Reasoning.Setoid as EqR                                              
open Magma M using (_≈_; _∙_; setoid) renaming (Carrier to A; refl to ≈refl)                
open Magma M' using () renaming ...                                                         
open Definitions A B _≈'_ using (Homomorphic₂)                                              
open Definitions B A _≈_  using () renaming (Homomorphic₂ to Homomorphic₂')                 
open EqR setoid                                                                             
...                                                                           
```                                                                                                                                            
But you see, this is not natural: setting inside a parametric module an import              
that does not depend on the parameters of this module. This gives the reader a              
wrong impression of the dependences.

Reply:
> But you see, this is not natural: setting inside a parametric module an import
that does not depend on the parameters of this module.

I agree with you on this one, but if we want to change this we should change it consistently across the library. Feel free to open an issue.